mitodl/ocw-hugo-themes
# OCW Stories & Quotes New Design Implementation

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/1384
Please note that there is a lot discussion in the ticket, most of which is not relevant anymore. So you may want to just focus on the Figma designs for this PR.

### Description (What does it do?)
Implements redesign of OCW Stories & Quotes. It includes:
- Homepage
- Stories List Page 
- Stories Single Page

Figma designs: 
[Desktop](https://www.figma.com/proto/MJPoWIwYmmJO0gFbGcZTcp/OCW-Stories?page-id=1035%3A386&node-id=1035-438&starting-point-node-id=1035%3A438&scaling=min-zoom&content-scaling=fixed&t=YOKK0Xzmq2MUBxcn-1)

[Homepage](https://www.figma.com/proto/MJPoWIwYmmJO0gFbGcZTcp/OCW-Stories?page-id=1035%3A386&node-id=1035-438&node-type=frame&viewport=688%2C429%2C0.18&t=CQwi9vRTa9xAQi10-1&scaling=min-zoom&content-scaling=fixed&starting-point-node-id=1035%3A438) - Desktop
[Stories Listing page](https://www.figma.com/proto/MJPoWIwYmmJO0gFbGcZTcp/OCW-Stories?page-id=1035%3A386&node-id=1035-522&node-type=frame&viewport=688%2C429%2C0.18&t=CQwi9vRTa9xAQi10-1&scaling=min-zoom&content-scaling=fixed&starting-point-node-id=1035%3A438) - Desktop
[Story Detail page](https://www.figma.com/proto/MJPoWIwYmmJO0gFbGcZTcp/OCW-Stories?page-id=1035%3A386&node-id=1035-473&node-type=frame&viewport=688%2C429%2C0.18&t=CQwi9vRTa9xAQi10-1&scaling=min-zoom&content-scaling=fixed&starting-point-node-id=1035%3A438) - Desktop

[Home section](https://www.figma.com/proto/MJPoWIwYmmJO0gFbGcZTcp/OCW-Stories?page-id=1035%3A386&node-id=1400-554&node-type=frame&viewport=584%2C441%2C0.14&t=Or4UeHQtnx3pM7Er-1&scaling=min-zoom&content-scaling=fixed&starting-point-node-id=1400%3A554&show-proto-sidebar=1) - Mobile
[Stories listing page ](https://www.figma.com/proto/MJPoWIwYmmJO0gFbGcZTcp/OCW-Stories?page-id=1035%3A386&node-id=1400-642&node-type=frame&viewport=584%2C441%2C0.14&t=2pBevSc14lpf89Al-1&scaling=min-zoom&content-scaling=fixed&starting-point-node-id=1035%3A438&show-proto-sidebar=1)- Mobile - the only change is the header, rest of the story listing will remain same as the home cards.
[Story Detail page](https://www.figma.com/proto/MJPoWIwYmmJO0gFbGcZTcp/OCW-Stories?page-id=1035%3A386&node-id=1400-1104&node-type=frame&viewport=584%2C441%2C0.14&t=2pBevSc14lpf89Al-1&scaling=min-zoom&content-scaling=fixed&starting-point-node-id=1035%3A438&show-proto-sidebar=1) - Mobile


### How can this be tested?
Test the following links:

- [Homepage](http://localhost:3000/)
- [All Stories Page](http://localhost:3000/stories/)
- [Random Story Example](http://localhost:3000/stories/demetra-tzevelekos/)

1. Checkout to this branch. 
2. `yarn start www`
3. Make sure things are working as expected in the Figma designs. 
4. Test the changes across all screen sizes to confirm they are responsive and visually consistent. 
5. Ensure all homepage features in www theme are working as before, especially carousel in Featured Courses/New Courses/OCW Stories.
6. Although quotes functionality relies on backend implementation (which will be done shortly next), it can still be tested. Add the following fields to the frontmatter of a markdown file for a story:
```
is_quote: true
leadquote: Your quote here
```
7. Ensure the Quotes Modal opens and functions as intended.

### Screenshots
I made some minor padding/margin related changes after taking these screenshots. The overall look is more or less same. But for exact changes, please refer to the PR deployment link: https://ocw-hugo-themes-www-pr-1460--ocw-next.netlify.app/

### Desktop screenshots
Single page: 
<img width="968" alt="image" src="https://github.com/user-attachments/assets/ee1e7a4c-5d87-4ac2-bd8a-cdf6ad275517">
<img width="971" alt="image" src="https://github.com/user-attachments/assets/91e0f4b8-3e93-41b5-8950-ea57917c9101">
(There are 4 cards on right)

List page:
<img width="962" alt="image" src="https://github.com/user-attachments/assets/c79963d6-ab99-4230-90e1-4b5a6cdf3de9">
<img width="974" alt="image" src="https://github.com/user-attachments/assets/79344c24-3971-4945-9dd6-fa873e963150">

Homepage:
<img width="866" alt="image" src="https://github.com/user-attachments/assets/ff1b74cd-64e0-4104-9867-c0f208d87736">

Quotes card:
<img width="794" alt="image" src="https://github.com/user-attachments/assets/d4c79abc-3175-4864-8d3e-84c5b7857335">


### Medium screen:
Homepage:
<img width="634" alt="image" src="https://github.com/user-attachments/assets/39726980-9096-4e8e-a1c9-57644639bc44">

List page:
<img width="534" alt="image" src="https://github.com/user-attachments/assets/cab6d6e3-3f7c-4b83-8fd9-10f774226e09">
<img width="532" alt="image" src="https://github.com/user-attachments/assets/9e404665-6e51-4d11-a497-8c72aba0ec1c">

Single page:
<img width="534" alt="image" src="https://github.com/user-attachments/assets/2da783b4-b149-49e6-a826-8c7a9d05fd2d">
<img width="560" alt="image" src="https://github.com/user-attachments/assets/dc3e6fa1-f031-47fe-8d2d-552071a583b6">

Quotes card:
<img width="585" alt="image" src="https://github.com/user-attachments/assets/0daeaad4-4443-4d36-8b8e-4b38749ecd33">

### Small screen:
Quotes card:
<img width="480" alt="image" src="https://github.com/user-attachments/assets/df982715-8981-4082-a5f9-06aa1850ff50">

Homepage:
<img width="465" alt="image" src="https://github.com/user-attachments/assets/ae7b084a-41e2-4968-b0d5-9ca114757684">

Single page:
<img width="319" alt="image" src="https://github.com/user-attachments/assets/faa88c0f-75d3-48d6-9a97-d60fa07f6728">
<img width="515" alt="image" src="https://github.com/user-attachments/assets/d5a7d1c6-dbff-4a57-b89b-fc66ecf2c4c2">

List page:
<img width="523" alt="image" src="https://github.com/user-attachments/assets/fd6b1adf-1ee8-4604-aad7-cf5ce646d758">
<img width="510" alt="image" src="https://github.com/user-attachments/assets/5e5a1a71-aa9b-4f6a-9cd5-fe074036103a">
